lisp/ess-inf.el

```diff
@@ -581,21 +574,30 @@ This marks the process with a message, at a particular time point."
 
 ;;*;; Requester functions called at startup
 
-(defun ess-get-directory (default dialect procname)
-  (let ((prog-version (cond ((string= dialect "R")
-                             (concat ", " inferior-R-version)) ; notably for the R-X.Y versions
-                            (inferior-ess-program
-                             (concat ", " inferior-ess-program ))
-                            (t ""))))
-    (ess-prompt-for-directory
-     (directory-file-name default)
-     (format "ESS (*%s*%s) starting data directory? "
-             procname prog-version)
-     ;; (format "ESS [%s {%s(%s)}: '%s'] starting data directory? "
-     ;;         ;;FIXME: maybe rather tmp-dialect (+ evt drop ess-language?)?
-     ;;         procname ess-language ess-dialect prog-version)
-     )))
-
+;; FIXME EMACS 25.1:
+;; Deprecate `ess-directory-function' in favour of `project-find-functions'?
+(defun inferior-ess--default-directory ()
+  (let ((dir (or (and ess-directory-function
+                      (funcall ess-directory-function))
+                 ess-default-directory
+                 default-directory)))
+    (directory-file-name dir)))
+
+(defun inferior-ess--get-directory (procname dialect)
+  "This returns the directory of the current project"
```

Comment:
Nitpicking, but let's do new docs according to emacs standards (imperative statement + properly document all params).

The name is not particularly suggestive. What directory does this return more concretely?  Shouldn't this just be part of `ess-prompt-for-directory`? It's a bit strange that a helper `--` function calls main utility function and not the other way around.

Reply:
>  let's do new docs according to emacs standards

Agreed. That was a leftover from the time where this was a `(def-generic ess-current-project ()`. I'll just remove the docstring.

> The name is not particularly suggestive.

Agreed, it should probably be `inferior-ess--get-startup-directory`. I don't like that function either and thought about merging it in `ess-prompt-for-directory` but it's a bit annoying because of that dialect specific handling. Also this checks for the value `ess-ask-for-ess-directory` to avoid multiple check in `inferior-ess-mode`.

> It's a bit strange that a helper -- function calls main utility function and not the other way around.

I think that's alright if it reduces repetition in another main function.